src/shared/components/VideoPlayer/VideoPlayer.tsx

```diff
@@ -115,37 +158,201 @@ const VideoPlayerComponent: React.ForwardRefRenderFunction<HTMLVideoElement, Vid
     player.play()
   }
 
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+    const handler = () => setVideoTime(Math.floor(player.currentTime()))
+
+    player.on('timeupdate', handler)
+    return () => {
+      player.off('timeupdate', handler)
+    }
+  }, [player])
+
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+    const handler = () => {
+      if (player.isFullscreen()) {
+        setIsFullScreen(true)
+      } else {
+        setIsFullScreen(false)
+      }
+    }
+    player.on('fullscreenchange', handler)
+    return () => {
+      player.off('fullscreenchange', handler)
+    }
+  }, [player])
+
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+    const handler = () => {
+      if (player.isFullscreen()) {
+        setIsFullScreen(true)
+      } else {
+        setIsFullScreen(false)
+      }
+    }
+    player.on('fullscreenchange', handler)
+    return () => {
+      player.off('fullscreenchange', handler)
+    }
+  }, [player])
+
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+    const enterPiPhandler = () => {
+      setIsPiPEnabled(true)
+    }
+    const exitPiPhandler = () => {
+      setIsPiPEnabled(false)
+    }
+
+    player.on('enterpictureinpicture', enterPiPhandler)
+    player.on('leavepictureinpicture', exitPiPhandler)
+    return () => {
+      player.off('enterpictureinpicture', enterPiPhandler)
+      player.off('leavepictureinpicture', exitPiPhandler)
+    }
+  }, [player])
```

Comment:
I think we should leave a little comment on top of useEffects just to point out what is it that the effect is for a lot of times it's hard to know what is going on